rpc/wrtc_base_channel.go

```diff
@@ -67,6 +67,15 @@ func newBaseChannel(
 		// to receive messages when the peer is done.
 		ch.cancel()
 
+		// RSDK-9969: Go through `webrtcBaseChannel.Close` before `onPeerDone` which may
+		// (additionally) call `PeerConnection.GracefulClose`. `webrtcBaseChannel.Close` will, among
+		// other things, poke condition variables with the appropriate locks to ensure that any
+		// goroutines blocked on sending data/servicing a request will wake up and exit. Returning
+		// their resources (e.g: call ticket).
+		//
+		// See RSDK-9969 for reproduction steps.
+		ch.Close()
+
 		if !peerDoneOnce && onPeerDone != nil {
 			peerDoneOnce = true
 			onPeerDone()
```

Comment:
Can't comment on the line, but I think we'll need a _bit_ more logic about marking `ch.activeBackgroundWorkers.Done()` _before_ calling `doPeerDone()`. I found a change that's ugly but works; let's discuss offline.

Reply:
You're right this was broken because ch.Close waits on background workers. And this ice state change handler was considered a background worker.

I changed the ice state change handler to not kick off a background worker for this disconnecting case (in addition to many of the other cases).